spring-batch-integration/src/main/java/org/springframework/batch/integration/partition/MessageChannelPartitionHandler.java

```diff
@@ -213,7 +213,7 @@ public void setReplyChannel(PollableChannel replyChannel) {
 		final Set<StepExecution> split = stepExecutionSplitter.split(masterStepExecution, gridSize);
 
 		if(CollectionUtils.isEmpty(split)) {
-			return null;
+			return split;

```

Comment:
This caused a test to fail.  `MessageChannelPartitionHandlerTests#testNoPartitions()`